server/src/main/java/io/crate/planner/operators/JoinPlanBuilder.java

```diff
@@ -236,17 +208,16 @@ private static LogicalPlan joinWithNext(Function<AnalyzedRelation, LogicalPlan>
                 queryParts.remove(Collections.singleton(nextName)))
                 .filter(Objects::nonNull).iterator()
         );
-        return Filter.create(
-            createJoinPlan(
-                source,
-                nextPlan,
-                type,
-                condition,
-                leftRelation,
-                query,
-                hashJoinEnabled),
-            query
-        );
+        boolean isFiltered = query.symbolType().isValueSymbol() == false;
+        var joinPlan = new NestedLoopJoin(
+            source,
+            nextPlan,
+            type,
+            condition,
+            isFiltered,
+            leftRelation,
+            isFiltered);
```

Comment:
Double checking, is this correct? because previously, the `createJoinPlan` here was using always false.

Reply:
Good catch, thanks. This is wrong. Last property should be false.